propka/atom.py

```diff
@@ -92,104 +101,148 @@ def set_properties(self, line):
+            if ba.element == element:
+                res.append(ba)
         return res
 
     def get_bonded_heavy_atoms(self):
-        return [ba for ba in self.bonded_atoms if ba.element!='H']
+        """Get the atoms bonded to this one that aren't hydrogen.
 
+        TODO - this could be a @property attribute/method
+
+        Returns:
+            list of atoms.
+        """
+        array = [ba for ba in self.bonded_atoms if ba.element != 'H']
+        return array
 
     def is_atom_within_bond_distance(self, other_atom, max_bonds, cur_bond):
-        """ check if <other_atom> is found within <max_bonds> bonds of self """
+        """Check if <other_atom> is found within <max_bonds> bonds of self.
+
+        Args:
+            other_atom:  atom to check
+            max_bonds:  number of bonds to check for other atom bonding to self
+        Returns:
+            Boolean for atom bond distance
+        """
         for ba in self.bonded_atoms:
             if ba == other_atom:
                 return True
             if max_bonds > cur_bond:
                 if ba.is_atom_within_bond_distance(other_atom, max_bonds, cur_bond+1):
                     return True
-
         return False
 
-
-
-    def setProperty(self,
-                    numb    = None,
-                    name    = None,
-                    resName = None,
-                    chainID = None,
-                    resNumb = None,
-                    x       = None,
-                    y       = None,
-                    z       = None,
-                    occ     = None,
-                    beta    = None):
+    def set_property(self, numb=None, name=None, res_name=None, chain_id=None,
+                     res_num=None, x=None, y=None, z=None, occ=None, beta=None):
+        """Set properties of the atom object.
+
+        Args:
+            numb:  Atom number
+            name:  Atom name
+            res_name:  residue name
+            chain_id:  chain ID
+            res_num:  residue number
+            x:  atom x-coordinate
+            y:  atom y-coordinate
+            z:  atom z-coordinate
+            occ:  atom occupancy
+            beta:  atom temperature factor
         """
-        sets properties of the atom object
+        if numb is not None:
+            self.numb = numb
+        if name is not None:
+            self.name = name
+        if res_name is not None:
+            self.res_name = res_name
+        if chain_id is not None:
+            self.chain_id = chain_id
+        if res_num is not None:
+            self.res_num = res_num
+        if x is not None:
+            self.x = x
+        if y is not None:
+            self.y = y
+        if z is not None:
+            self.z = z
+        if occ is not None:
+            self.occ = occ
+        if beta is not None:
+            self.beta = beta
+
+    def make_copy(self):
+        """Make a copy of this atom.
+
+        TODO - this could be a property method/attribute
```

Comment:
I would leave it a method because it produces something new every time you call it. A property masks this and makes the user think that this is static content.

Reply:
Addressed by https://github.com/Electrostatics/propka-3.1/commit/120718fbd8e385c5ef734b825252f1d57ea17e2c